sklearn/utils/_clibs.py

```diff
@@ -0,0 +1,411 @@
+"""
+This module provides utilities to load C-libraries that relies on thread
+pools and limit the maximal number of thread that can be used.
+"""
+
+# This code is adapted from code by Thomas Moreau <tomMoral> available at
+# https://github.com/tomMoral/loky
```

Comment:
Should this go to `externals` ?

Ideally, in the long term I think we want to use this functionality from loky  ?

Reply:
Yes but it's there is a broken test in loky at the moment, so it's not released there. It might even belong to its own project. But for now I am fine with keeping a local copy here. Once it's also available elsewhere we can discuss vendoring vs adding a dependency.

Reply:
The failing test is probably the same as the one @jeremiedbb just fixed. I will try fixing the PR now and see how it goes.

Reply:
I am fine with staying in utils for now.